src/functions.lisp

```diff
@@ -162,18 +176,17 @@
          (append fixed-fargs varargs-fargs)
          (append fixed-types varargs-types)
          rettype
-         `(,(if pointerp '%foreign-funcall-pointer '%foreign-funcall)
+         `(,(if pointerp '%foreign-funcall-pointer-varargs '%foreign-funcall-varargs)
             ,thing
+            ,(mapcan #'list fixed-ctypes fixed-syms)
             ,(append
               (mapcan #'list
-                      (nconc fixed-ctypes
-                             (mapcar #'promote-varargs-type varargs-ctypes))
-                      (append fixed-syms
-                              (loop for sym in varargs-syms
-                                    and type in varargs-ctypes
-                                    if (eq type :float)
-                                    collect `(float ,sym 1.0d0)
-                                    else collect sym)))
+                      (mapcar #'promote-varargs-type varargs-ctypes)
+                      (loop for sym in varargs-syms
+                         and type in varargs-ctypes
+                         if (eq type :float)
+                         collect `(float ,sym 1.0d0)
+                         else collect sym))
```

Comment:
Did a little bit of bikeshedding here to indent the loop the way I like it: f696f3db4ac75e6a516e3a238a28cf1c327776f2.